ui/app/components/calendar-widget.js

```diff
@@ -0,0 +1,176 @@
+import Component from '@glimmer/component';
+import layout from '../templates/components/calendar-widget';
+import { setComponentTemplate } from '@ember/component';
+import { format, formatRFC3339 } from 'date-fns';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+// ARG TODO documentation takes start and end for handQuery
+/**
+ * @module CalendarWidget
+ * CalendarWidget components are used in the client counts metrics. It helps users understand the ranges they can select.
+ *
+ * @example
+ * ```js
+ * <CalendarWidget
+ * @param {function} handleQuery - calls the parent pricing-metrics-dates handleQueryFromCalendar method which sends the data for the network request.
+ * @param {object} startDate - The start date is calculated from the parent. This component is only responsible for modifying the end Date. ANd effecting single month.
+ * />
+ *
+ * ```
+ */
+
+const MONTH_MAPPING = [
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December',
+];
+
+class CalendarWidget extends Component {
+  currentDate = new Date();
+  currentYear = this.currentDate.getFullYear(); // integer
+  currentMonth = parseInt(format(this.currentDate, 'M')) - 1; // integer and zero index
+
+  @tracked allMonthsNodeList = [];
+  @tracked displayYear = this.currentYear; // init to currentYear and then changes as a user clicks on the chevrons
+  @tracked disablePastYear = this.isObsoleteYear(); // if obsolete year, disable left chevron
+  @tracked disableFutureYear = this.isCurrentYear(); // if current year, disable right chevron
+  @tracked endDateDisplay;
+  @tracked showCalendar = false;
+  @tracked showSingleMonth = false;
+
+  constructor() {
+    super(...arguments);
+    // ARG TODO we need to return the config's duration if not default to 12 months to calculate, now it's 12 months
+    let date = new Date();
+    date.setMonth(date.getMonth() - 1); // by default you calculate the end month as the month prior to the current month.
+    this.endDateDisplay = format(date, 'MMMM yyyy');
```

Comment:
I think naming here is a little confusing (yay dates!) but this will be the `startDateDisplay` as it's the billing start date and, as you mentioned above, will be grabbed from the license endpoint? 

Once all wired up, the constructor may look more like this - right? 
```
    let startDate = this.args.billingStart
    this.startDateDisplay = format(startDate, 'MMMM yyyy') 
```

Reply:
Good question. I think my comment needs to change. Because it's the endDate that is displayed. 
![image](https://user-images.githubusercontent.com/6618863/148833165-dfbc5fa9-fb53-4075-9e76-5881ad3535cd.png)
The Decemeber 2021 in this image.

Reply:
Ah! I see - yes that will change/be clearer with real data